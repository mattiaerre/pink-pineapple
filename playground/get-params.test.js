const getParams = require('./get-params');

const scenarios = [
  {
    description: 'oc-apod',
    parameters: {
      apiKey:
      {
        type: 'string',
        mandatory: true,
        example: 'DEMO_KEY',
        description: 'The NASA Open APIs key'
      }
    }
  },
  {
    description: 'oc-columbus-header',
    parameters: {
      title:
      {
        type: 'string',
        mandatory: true,
        example: 'Instagram',
        description: 'The main title'
      },
      logoUrl:
      {
        type: 'string',
        mandatory: true,
        example: 'http://www.uidownload.com/files/722/15/621/facebook-instagram-instagram-2016-instagram-logo-new-new-instagram-icon.png',
        description: 'The logo\'s absolute url'
      }
    }
  },
  {
    description: 'oc-social-media-buttons',
    parameters: undefined
  },
  {
    description: 'oc-timeago',
    parameters: {
      start:
      {
        type: 'string',
        mandatory: true,
        example: '2016-02-22',
        description: 'The start date as string in a YYYY-MM-DD format'
      },
      end:
      {
        type: 'string',
        mandatory: false,
        example: '2017-02-22',
        description: 'The end date as string in a YYYY-MM-DD format'
      }
    }
  }
];

scenarios.forEach((scenario) => {
  test(scenario.description, () => {
    const component = {
      oc: {
        parameters: scenario.parameters
      }
    };

    const params = getParams(component);
    expect(params).toMatchSnapshot();
  });
});
