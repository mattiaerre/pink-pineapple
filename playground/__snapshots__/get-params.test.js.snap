// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`oc-apod 1`] = `
Object {
  "apiKey": "DEMO_KEY",
}
`;

exports[`oc-columbus-header 1`] = `
Object {
  "logoUrl": "http://www.uidownload.com/files/722/15/621/facebook-instagram-instagram-2016-instagram-logo-new-new-instagram-icon.png",
  "title": "Instagram",
}
`;

exports[`oc-social-media-buttons 1`] = `Object {}`;

exports[`oc-timeago 1`] = `
Object {
  "start": "2016-02-22",
}
`;
